bridge/whatsapp/whatsapp.go

```diff
@@ -266,18 +269,70 @@ func (b *Bwhatsapp) Send(msg config.Message) (string, error) {
 		// TODO handle edit as a message reply with updated text
 	}
 
-	//// TODO Handle Upload a file
-	//if msg.Extra != nil {
-	//	for _, rmsg := range helper.HandleExtra(&msg, b.General) {
-	//		b.c.SendMessage(roomID, rmsg.Username+rmsg.Text)
-	//	}
-	//	if len(msg.Extra["file"]) > 0 {
-	//		return b.handleUploadFile(&msg, roomID)
-	//	}
-	//}
+	// Handle Upload a file
+	if msg.Extra["file"] != nil {
+		fi := msg.Extra["file"][0].(config.FileInfo)
+
+		filetype := mime.TypeByExtension(filepath.Ext(fi.Name))
+
+		b.Log.Debugf("Extra file is %#v", filetype)
+
+		// TODO: add different types
+		// TODO: add webp conversion
+		if filetype == "image/jpeg" || filetype == "image/png" || filetype == "image/gif" {
```

Comment:
If we are going to handle different types I'd suggest moving this to a `switch` statement instead and split out the type-specific handling into a separate function. That way you can also pull out the current "default" code (below this `if` statement) into the `default` case of the `switch` and it's own function. In the current state it would be the same function as the two code-paths are identical which will reduce code deduplication.

This should all help improve the cleaness and readability of the code. :)